### PR TITLE
add signing - FLAT-2381

### PR DESCRIPTION
**Pull request by ronanbarrett**
** do not merge until all the calling actions have been updated **

all repos migrated..let's go!

### CODE REVIEW

File: argo-cd-release-production/action.yml [hunk 1]
```diff
@@ -7,15 +7,34 @@ inputs:
...
     description: 'Path where the manifests live'
     required: false
     default: '.'
 
 runs:
   using: "composite"
-
   steps:
+    - name: Start SSH agent and load private key for commit signing
+      uses: webfactory/ssh-agent@v0.5.4
```
Comment (K-Phoen) [inline argo-cd-release-production/action.yml:35]: There's a more recent version of this action, that doesn't use a deprecated nodejs runtime :)

File: argo-cd-release-production/action.yml [hunk 2]
```diff
@@ -46,8 +65,12 @@ runs:
 
         IMAGE=$(yq r ${VALUES_FILE} "image" | sed "s/:.*$//")":${VERSION}"
 
-        git config --global user.email ${GITHUB_ACTOR}@voiapp.io
-        git config --global user.name ${GITHUB_ACTOR}
+        git config --global user.email "${{ inputs.bot_email }}"
+        git config --global user.name "${{ inputs.bot_name }}"
+        git config --global gpg.format "ssh"
+        git config --global user.signingkey '${{ inputs.bot_ssh_public_key }}'
+        git config --global commit.gpgsign "true"
```
Comment (K-Phoen) [inline argo-cd-release-production/action.yml:72]: Is that enough?

In voictl (or locally), we also have to add the SSH key into the "allowed signers" file used by git: https://github.com/voiapp/voictl/blob/993eeb8d437bc2e72788393f7349720f5fb072a0/cmd/voictl/commands/setup/commitsigning.go#L94

File: build-and-release-to-env/action.yaml [hunk 3]
```diff
@@ -110,13 +126,17 @@ runs:
...
         if test -n "${{ inputs.pr-number }}"; then
-          TAG_NAME=PR-"${{ inputs.pr-number }}" make push
+          export TAG_NAME=PR-"${{ inputs.pr-number }}"
         else
-          DOCKER_TOOLS=true TAG_NAME=$(basename ${GITHUB_REF}) make push
+          export DOCKER_TOOLS=true TAG_NAME=$(basename ${GITHUB_REF})
         fi
 
+        make push
+        echo "version=${TAG_NAME}" >> $GITHUB_OUTPUT
```
Comment (K-Phoen) [inline build-and-release-to-env/action.yaml:138]: Is that useful?

File: argo-cd-release-production/action.yml (hunk 2 above)
Comment (IuryAlves) [reply argo-cd-release-production/action.yml:72]: It worked in our tests, but it doesn't hurt to add one extra line.

Comment (K-Phoen) [reply argo-cd-release-production/action.yml:72]: If it's not needed, it's fine :shrug:

File: build-and-release-to-env/action.yaml (hunk 3 above)
Comment (IuryAlves) [reply build-and-release-to-env/action.yaml]: Removed

File: argo-cd-release-production/action.yml (hunk 2 above)
Comment (IuryAlves) [reply argo-cd-release-production/action.yml:72]: Fixed!

File: argo-cd-release-production/action.yml (hunk 1 above)
Comment (IuryAlves) [reply argo-cd-release-production/action.yml]: Fixed

**Comment by IuryAlves** [comment]
All comments fixed @K-Phoen

**Review by K-Phoen — APPROVED** [review]

**Pull request merged**